fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java

```diff
@@ -577,7 +568,16 @@ private void insertRdfTypes(final List<URI> rdfTypes) {
             parameterSourcesList.add(assocParams);
         }
         final MapSqlParameterSource[] psArray = parameterSourcesList.toArray(new MapSqlParameterSource[0]);
-        jdbcTemplate.batchUpdate(INSERT_RDF_TYPE, psArray);
+
+        // RDF types are upserted outside of the tx to avoid concurrent update contention between txs, especially
+        // in MySQL. This means that they will not be rolled back if the tx fails.
+        noTransactionTemplate.executeWithoutResult(status -> {
+            try {
+                jdbcTemplate.batchUpdate(INSERT_RDF_TYPE, psArray);
```

Comment:
@pwinckles :  if the first item in the array fails due to a DuplicateKeyException will subsequent arrays be inserted?

Reply:
Ah, good catch. I forgot that was a batch update